File: src/main/java/org/chousik/collection/validators/ValidatorLocationX.java
package org.chousik.collection.validators;

/**
 * Валидатор координаты X
 */
public class ValidatorLocationX implements IValidator<Double> {
    /**
     * Проверяет число на соответствие диапазону
     *
     * @param aDouble число
     * @throws NullPointerException если число null
     */
    @Override
    public void valide(Double aDouble) throws NullPointerException {
        if (aDouble == null | aDouble.isNaN()) {
            throw new NullPointerException();
        }
    }
}

File: src/main/kotlin/collection/validators/ValidatorLocationX.kt
package org.chousik.collection.validators


class ValidatorLocationX : IValidator<Double?> {

    @Throws(NullPointerException::class)
    override fun valide(t: Double?) {
        if ((t == null) or t!!.isNaN()) {
            throw NullPointerException()
        }
    }
}